### PR TITLE
Add integration tests with Docker image

### PR DESCRIPTION
**Pull request by daisy1754**
# Description
This PR adds integration test backed by Docker. 

- `.circleci/docker/integration-tests/Dockerfile` defines our own Docker image named `boltdev/m1-plugin-ci-php56-mage19`. It comes with php5.6, apache, magento 1.9, and MySQL
- During test run, we install magento sample data, install Bolt plugin and setup some seed data (two new scripts added under tests/script)
- Then we setup ngrok and run integration test against local magento instance

Fixes: https://boltpay.atlassian.net/browse/M1P-18

#changelog Add integration tests with Docker image

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: tests/scripts/setup_integration.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+while ! sudo mysql -uroot -h localhost -e "SELECT 1" >/dev/null 2>&1; do
+    sleep 1
+done
+
+sudo mysql -u root -h localhost -e "CREATE USER 'magento'@'127.0.0.1'"
+sudo mysql -u root -h localhost -e "GRANT ALL PRIVILEGES ON *.* TO 'magento'@'127.0.0.1' WITH GRANT OPTION"
+sudo mysql -u root -h localhost -e "FLUSH PRIVILEGES"
+mysql -u magento -h 127.0.0.1 -e "CREATE DATABASE IF NOT EXISTS magento"
+mysql -u magento -h 127.0.0.1 magento < magento-sample-data-1.9.2.4/magento_sample_data_for_1.9.2.4.sql
+php -d memroy_limit=512M n98-magerun.phar install --magentoVersionByName=magento-mirror-1.9.3.6 \
```
Comment (roopakv) [inline tests/scripts/setup_integration.sh:24]:
```suggestion
php -d memory_limit=512M n98-magerun.phar install --magentoVersionByName=magento-mirror-1.9.3.6 \
```

File: tests/scripts/setup_integration.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  --installationFolder=/var/www/html --forceUseDb --noDownload --dbHost=127.0.0.1 --dbUser=magento --dbName=magento  --dbPass="" \
+  --installSampleData=yes --useDefaultConfigParams=yes --baseUrl=http://m1-test.integrations.dev.bolt.me
+php -d memroy_limit=512M n98-magerun.phar admin:user:create bolttest dev+m1-integration-admin@bolt.com bolt1234 \
```
Comment (roopakv) [inline tests/scripts/setup_integration.sh:27]:
```suggestion
php -d memory_limit=512M n98-magerun.phar admin:user:create bolttest dev+m1-integration-admin@bolt.com bolt1234 \
```

File: .circleci/docker/integration-tests/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+RUN perl -pi -e "s/mailhub=mail/mailhub=maildev/" /etc/ssmtp/ssmtp.conf
+
+RUN a2enmod rewrite
+
+# Download magento
+USER circleci
+WORKDIR /home/circleci
+RUN curl -O https://files.magerun.net/n98-magerun.phar
+RUN chmod +x n98-magerun.phar
+RUN php -d memroy_limit=512M n98-magerun.phar install --magentoVersionByName=magento-mirror-1.9.3.6 \
```
Comment (matt-thomason) [inline .circleci/docker/integration-tests/Dockerfile:109]: Another `memory_limit`

**Comment by adentong** [comment]
i found another memroy (even though it's not part of this PR), but i'm fixing it anyways

**Comment by adentong** [comment]
Also @daisy1754 this is pretty insane given this is a mere side project :P

**Pull request merged**